spec/integration/rom/factory_spec.rb

```diff
@@ -899,6 +899,36 @@ class Admin < ROM::Struct
       end
     end
 
+    context "has_many-through" do
+      before do
+        factories.define(:user) do |f|
+          f.first_name "Jane"
+          f.last_name "Doe"
+          f.email "jane@doe.org"
+          f.timestamps
+          f.association(:addresses, count: 2)
+        end
+
+        factories.define(:address) do |f|
+          f.sequence(:full_address) { |n| "Address #{n}" }
+        end
+      end
+
+      it "creates associated records" do
```

Comment:
1. When the intermidiate association table has `Not Null` constraint for timestamps,  it'll fail. Can we use factory (if exists) for intermidiate record creation (in this case, it's the `user_addresses`

2. This does not allow us to build structs without persisting (You'll get missing user_id in Hash Input